phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/common/widgets/details/skin/DetailsListWidgetSkin.java

```diff
@@ -100,7 +100,10 @@ private void updateOldSelection(ListView<DetailsListElement<E>> container, ListW
         Optional.ofNullable(oldSelection).map(ListWidgetSelection::getSelection).ifPresent(selection -> {
             final int oldValueIndex = getControl().getElements().indexOf(selection);
 
-            container.getSelectionModel().clearSelection(oldValueIndex);
+            // In some corner cases (category / filter changes, the current selection can disappear)
+            if (oldValueIndex != -1) {
```

Comment:
I think all three skins require both `mappedElements.size() > oldValueIndex && oldValueIndex != -1`.
I think `container.getSelectionModel().clearSelection(oldValueIndex)` will throw a `NullPointerException` if the index is larger than the number of shown elements.

Reply:
In fact I'm not totally sure. The code is the the same

Reply:
Which code is the same?

Reply:
is not*

Reply:
I'm not sure I understand.
The main difference is that the `DetailsListWidget` and the `CompactListWidget` delegate the deselection to the internal `SelectionModel`, while the `IconsListWidget` provides its own logic for the selection and deselection of items.
Both approaches require a valid `oldValueIndex` which is in the interval `[0, mappedElements.size()]` to work correctly.